backend/tests/Feature/SubmissionTest.php

```diff
@@ -0,0 +1,248 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\Feature;
+
+use App\Models\Publication;
+use App\Models\Role;
+use App\Models\Submission;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Nuwave\Lighthouse\Testing\MakesGraphQLRequests;
+use Tests\TestCase;
+
+class SubmissionTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
+
+    /**
+     * @return void
+     */
+    public function testSubmissionsHaveAOneToManyRelationshipWithPublications()
+    {
+        $publication = Publication::factory()->create([
+            'name' => 'Test Publication #1',
+        ]);
+        Publication::factory()->count(5)->create();
+        $submissions = Submission::factory()->count(10)->for($publication)->create();
+        Submission::factory()->count(16)->create();
+        $this->assertEquals(1, $submissions->pluck('publication_id')->unique()->count());
+        $this->assertGreaterThanOrEqual(10, $publication->submissions->count());
+        $this->assertLessThanOrEqual(26, $publication->submissions->count());
+    }
+
+    /**
+     * @return void
+     */
+    public function testSubmissionsHaveAManyToManyRelationshipWithUsers()
+    {
+        $submission_count = 4;
+        $user_count = 6;
+        $submitter_id = Role::where('name', Role::SUBMITTER)->first()->id;
+        $publication = Publication::factory()->create([
+            'name' => 'Test Publication #2',
+        ]);
+        $users = User::factory()->count($user_count)->create();
+
+        // Create submissions and attach them to users randomly with random roles
+        for ($i = 0; $i < $submission_count; $i++) {
+            $random_role_id = Role::whereIn(
+                'name',
+                [
+                    Role::REVIEW_COORDINATOR,
+                    Role::REVIEWER,
+                    Role::SUBMITTER,
+                ]
+            )
+                ->get()
+                ->pluck('id')
+                ->random();
+            $submission = Submission::factory()->hasAttached(
+                $users->random(),
+                [
+                    'role_id' => $random_role_id,
+                ]
+            )
+                ->for($publication)
+                ->create();
+
+            // Ensure at least one Submitter is attached if one was not previously attached
+            if ($random_role_id !== $submitter_id) {
+                $submission->users()->attach(
+                    $users->random(),
+                    [
+                        'role_id' => $submitter_id,
+                    ]
+                );
+            }
+        }
+        Submission::all()->map(function ($submission) use ($user_count) {
```

Comment:
- Should we add a test to check if the user for a submission is present in the users table?
- Should we add a test to check if the role for a submission is present in the roles table?

Reply:
Added the suggested assertions